servers/rendering/shader_preprocessor.cpp

```diff
@@ -1226,6 +1271,10 @@ Error ShaderPreprocessor::preprocess(State *p_state, const String &p_code, Strin
 				state->cyclic_include_hashes.erase(code_hash); // Remove this hash.
 				return FAILED;
 			}
+			// Remove any indendentation in front of the directive
+			if (output.size() > 0 && output[output.size() - 1] == ' ') {
```

Comment:
I don't think this is a good place to try to format the output to make testing easier, it would be an endless whack-a-mole situation. You can remove this and adjust the testing to ignore lines that only have whitespace or whatever is needed, you can be relaxed like that in the testing comparison. The empty lines are kept to make side-by-side diffing easier for debugging. The spaces are there because get_token() will collapse any sequence of whitespace (spaces, tabs) into a single space and this is not the only place which can cause that, for example this should also cause similar issues:

``` glsl
#if DEBUG
    #define MEMBER_DEBUG_VARIABLE float debug_value;
#else
    #define MEMBER_DEBUG_VARIABLE
#endif

struct Data {
    float x;
    MEMBER_DEBUG_VARIABLE
};
```

Reply:
Removed it and made the test comparator more lenient.